modules/Bio/EnsEMBL/Variation/DBSQL/PhenotypeFeatureAdaptor.pm

```diff
@@ -1074,6 +1074,116 @@ sub fetch_all {
   
 }
 
+=head2 get_PhenotypeFeatures_by_location
+
+  Arg [1]    : int $seq_region_id
+  Arg [2]    : int $seq_region_start
+  Arg [3]    : int $seq_region_end
+  Description: Retrieves PhenotypeFeatures by genomic location
+               
+  Returntype : hashref of array of types => phenotype_feature 
+  Exceptions : none
+  Caller     : general
+
+=cut
+
+sub get_PhenotypeFeatures_by_location {
+  my $self = shift;
+  my $seq_region_id = shift;
+  my $seq_region_start = shift;
+  my $seq_region_end = shift;
+  my $source_id = shift;
+
+  throw("Cannot fetch attributes without seq region information") unless defined($seq_region_id) && defined($seq_region_start) && defined($seq_region_end);
+
+  my $attribs = {};
+
+  my $extra_sql = " pf.seq_region_id = $seq_region_id AND pf.seq_region_start >= $seq_region_start AND pf.seq_region_end <= $seq_region_end ";
+  $extra_sql .= " AND pf.source_id = $source_id " if defined($source_id);
+
+  $extra_sql = $self->_is_significant_constraint($extra_sql);
+
+  return $self->generic_fetch("$extra_sql");
+}
+
+=head2 get_PhenotypeFeatureAttribs_by_location
```

Comment:
The name and docs on this method are confusing as it doesn't return all PhenotypeFeature attribs by location - it currently restricts to those with an associated allele. Names and docs should reflect function. 

We only use capitalisation like 'PhenotypeFeatureAttribs' when we are returning  PhenotypeFeatureAttrib objects, so it is not appropriate here as we don't have PhenotypeFeatureAttrib objects.

It would be clearer and more efficient to have a method  'get_clinsig_alleles_by_location' returning only alleles and their associated clinical significances per location. Then state in the docs that is has a specific caller.